src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacPropertyAttributes/MultipleAttributesOnDacPropertyFixBase.cs

```diff
@@ -0,0 +1,171 @@
+			Func<FieldTypeAttributeInfo, bool> removePredicate = GetRemoveAttributeByAttributeInfoPredicate();
+
+			if (codeActionName.IsNullOrWhiteSpace() || removePredicate == null)
+				return;
+
+			SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken)
+													.ConfigureAwait(false);
+
+			if (!(root?.FindNode(context.Span) is AttributeSyntax attributeNode) || context.CancellationToken.IsCancellationRequested)
+				return;
+
+			var propertyDeclaration = attributeNode.Parent<PropertyDeclarationSyntax>();
+
+			if (propertyDeclaration == null)
+				return;
+
+			SemanticModel semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken)
+																.ConfigureAwait(false);
+			if (semanticModel == null)
+				return;
+
+			CodeAction codeAction =
+				CodeAction.Create(codeActionName,
+								  cToken => RemoveAllOtherAttributesFromPropertyAsync(context.Document, root, attributeNode,
+																					  propertyDeclaration, semanticModel, 
+																					  removePredicate, cToken),
+								  equivalenceKey: codeActionName);
+
+			context.RegisterCodeFix(codeAction, context.Diagnostics);
+		}
+
+		protected abstract string GetCodeActionName();
+
+		protected abstract Func<FieldTypeAttributeInfo, bool> GetRemoveAttributeByAttributeInfoPredicate();
+
+		private Task<Document> RemoveAllOtherAttributesFromPropertyAsync(Document document, SyntaxNode root, AttributeSyntax attributeNode,
+																		 PropertyDeclarationSyntax propertyDeclaration, SemanticModel semanticModel,
+																		 Func<FieldTypeAttributeInfo, bool> removePredicate, CancellationToken cancellationToken)
+		{
+			return Task.Run(() =>
+			{
+				var rewriterWalker = new MultipleAttributesRemover(document, semanticModel, attributeNode, removePredicate, cancellationToken);
+				var propertyModified = rewriterWalker.Visit(propertyDeclaration) as PropertyDeclarationSyntax;
+
+				cancellationToken.ThrowIfCancellationRequested();
+
+				var modifiedRoot = root.ReplaceNode(propertyDeclaration, propertyModified);
+				return document.WithSyntaxRoot(modifiedRoot);
+
+			}, cancellationToken);
+		}
+
+
+
+		private class MultipleAttributesRemover : CSharpSyntaxRewriter
+		{
+			private int _visitedAttributeListCounter;
+			private int _attributeListsRemovedCounter;
+			
+			private readonly Document _document;
+			private readonly SemanticModel _semanticModel;
+			private readonly FieldTypeAttributesRegister _attributesRegister;
+			private readonly AttributeSyntax _remainingAttribute;
+			private readonly Func<FieldTypeAttributeInfo, bool> _attributeToRemovePredicate;
+			private readonly CancellationToken _cancellationToken;
+
+			public MultipleAttributesRemover(Document aDocument, SemanticModel aSemanticModel, AttributeSyntax aRemainingAttribute,
+											 Func<FieldTypeAttributeInfo, bool> attributeToRemovePredicate, CancellationToken cToken)
+			{
+				_document = aDocument;
```

Comment:
Unusual naming of method parameters - "a" prefix is not needed

Reply:
ok